Utilities/TestStreebog/StreebogPreprocess.c

```diff
@@ -0,0 +1,248 @@
+  Copyright (c) 2022 Maxim Kuznetsov. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause
+**/
+
+#include <UserFile.h>
+#include <Library/OcCryptoLib.h>
+#include <BigNumLib.h>
+
+CONST UINT8  M1[] = {
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32, 0x33,
+  0x34, 0x35,
+  0x36, 0x37,
+  0x38, 0x39,
+  0x30, 0x31,
+  0x32
+};
+
+CONST UINT8  M2[] = {
+  0xd1, 0xe5, 0x20, 0xe2,
+  0xe5, 0xf2, 0xf0, 0xe8,
+  0x2c, 0x20, 0xd1, 0xf2,
+  0xf0, 0xe8, 0xe1, 0xee,
+  0xe6, 0xe8, 0x20, 0xe2,
+  0xed, 0xf3, 0xf6, 0xe8,
+  0x2c, 0x20, 0xe2, 0xe5,
+  0xfe, 0xf2, 0xfa, 0x20,
+  0xf1, 0x20, 0xec, 0xee,
+  0xf0, 0xff, 0x20, 0xf1,
+  0xf2, 0xf0, 0xe5, 0xeb,
+  0xe0, 0xec, 0xe8, 0x20,
+  0xed, 0xe0, 0x20, 0xf5,
+  0xf0, 0xe0, 0xe1, 0xf0,
+  0xfb, 0xff, 0x20, 0xef,
+  0xeb, 0xfa, 0xea, 0xfb,
+  0x20, 0xc8, 0xe3, 0xee,
+  0xf0, 0xe5, 0xe2, 0xfb
+
+};
+
+CONST UINT8  H1_256[] = {
+  0x9d, 0x15,
+  0x1e, 0xef,
+  0xd8, 0x59,
+  0x0b, 0x89,
+  0xda, 0xa6,
+  0xba, 0x6c,
+  0xb7, 0x4a,
+  0xf9, 0x27,
+  0x5d, 0xd0,
+  0x51, 0x02,
+  0x6b, 0xb1,
+  0x49, 0xa4,
+  0x52, 0xfd,
+  0x84, 0xe5,
+  0xe5, 0x7b,
+  0x55, 0x00
+};
+
+CONST UINT8  H1_512[] = {
+  0x1b, 0x54,
+  0xd0, 0x1a,
+  0x4a, 0xf5,
+  0xb9, 0xd5,
+  0xcc, 0x3d,
+  0x86, 0xd6,
+  0x8d, 0x28,
+  0x54, 0x62,
+  0xb1, 0x9a,
+  0xbc, 0x24,
+  0x75, 0x22,
+  0x2f, 0x35,
+  0xc0, 0x85,
+  0x12, 0x2b,
+  0xe4, 0xba,
+  0x1f, 0xfa,
+  0x00, 0xad,
+  0x30, 0xf8,
+  0x76, 0x7b,
+  0x3a, 0x82,
+  0x38, 0x4c,
+  0x65, 0x74,
+  0xf0, 0x24,
+  0xc3, 0x11,
+  0xe2, 0xa4,
+  0x81, 0x33,
+  0x2b, 0x08,
+  0xef, 0x7f,
+  0x41, 0x79,
+  0x78, 0x91,
+  0xc1, 0x64,
+  0x6f, 0x48
+};
+
+CONST UINT8  H2_256[] = {
+  0x9d, 0xd2,
+  0xfe, 0x4e,
+  0x90, 0x40,
+  0x9e, 0x5d,
+  0xa8, 0x7f,
+  0x53, 0x97,
+  0x6d, 0x74,
+  0x05, 0xb0,
+  0xc0, 0xca,
+  0xc6, 0x28,
+  0xfc, 0x66,
+  0x9a, 0x74,
+  0x1d, 0x50,
+  0x06, 0x3c,
+  0x55, 0x7e,
+  0x8f, 0x50
+};
+
+CONST UINT8  H2_512[] = {
+  0x1e, 0x88,
+  0xe6, 0x22,
+  0x26, 0xbf,
+  0xca, 0x6f,
+  0x99, 0x94,
+  0xf1, 0xf2,
+  0xd5, 0x15,
+  0x69, 0xe0,
+  0xda, 0xf8,
+  0x47, 0x5a,
+  0x3b, 0x0f,
+  0xe6, 0x1a,
+  0x53, 0x00,
+  0xee, 0xe4,
+  0x6d, 0x96,
+  0x13, 0x76,
+  0x03, 0x5f,
+  0xe8, 0x35,
+  0x49, 0xad,
+  0xa2, 0xb8,
+  0x62, 0x0f,
+  0xcd, 0x7c,
+  0x49, 0x6c,
+  0xe5, 0xb3,
+  0x3f, 0x0c,
+  0xb9, 0xdd,
+  0xdc, 0x2b,
+  0x64, 0x60,
+  0x14, 0x3b,
+  0x03, 0xda,
+  0xba, 0xc9,
+  0xfb, 0x28
+};
+
+int
+ENTRY_POINT (
+  int   argc,
+  char  *argv[]
+  )
+{
+  UINT8  hash[64];
+  UINT8  TM2[72];
+  UINT8  Flag = 1;
+
+  Streebog256 (M1, 63, hash);
```

Comment:
Use sizeof when you refer to static arrays, not numbers.

Reply:
https://github.com/acidanthera/OpenCorePkg/pull/377/commits/e842c66f2450dc2d13c7da15f95f7c79c18391a8